dill/settings.py

```diff
@@ -21,5 +23,177 @@
+                    if not isinstance(filter, type):
+                        raise TypeError("filter is not a type: %r" % filter)
+                else:
+                    filter = 'type:%s' % name
+                module_filters.add(filter, rule_type=rule_type)
+        if '%s.funcs' % rule in section:
+            for code in section['%s.funcs' % rule].strip('\n').splitlines():
+                if code.startswith('(') and code.endswith(')'):
+                    code = code[1:-1]
+                globals_ = {}
+                if not code.startswith('lambda'):
+                    name = code.partition('(')[0]
+                    _resolve(name)  # import required modules
+                    top_level = name.partition('.')[0]
+                    globals_[top_level] = __import__(top_level)
+                filter = eval(code, globals_)
+                if not callable(filter):
+                    raise TypeError("filter is not callable: %r" % filter)
+                module_filters.add(filter, rule_type=rule_type)
+
+def read_settings(filename) -> None:
+    """Read dill settings from an INI file.
+
+    Update the ``dill.settings`` dictionary with the contents of the INI file
+    ``filename``.  Accepted file sections:
+
+      - `dill`: general :py:mod:`dill` settings
+      - `dill.module`: settings for :py:func:`dill.dump_module`
+      - `filters`: default exclude/include filters for :py:func:`dill.dump_module`
+      - `filters.<module_name>`: module-specific filters for
+        :py:func:`dill.dump_module`, where `<module_name>` is the complete module
+        path in the form `module[.submodule...]`
+
+    Accepted option values for general settings:
+
+      - boolean options (case insensitive): yes, no, on, off, true, false
+      - `protocol`: DEFAULT_PROTOCOL, HIGHEST_PROTOCOL, 0, 1, 2, 3, ...
+      - `fmode`: HANDLE_FMODE, 0, CONTENTS_FMODE, 1, FILE_FMODE, 2
+
+    .. IMPORTANT: The demo config file below is used in test_settings.py
+    .. Lexer 'pacmanconf' generates better highlighting than 'ini'.
+    .. code-block:: pacmanconf
+
+        [dill]
+        # General settings
+        ## Stored in dill.settings.
+        protocol = HIGHEST_PROTOCOL
+        byref = yes
+
+        [dill.session]
+        # Settings for dill.dump_module()
+        ## Stored in dill.session.settings.
+        refonfail = no
+
+        [filters]
+        # Default exclude/include filters for dill.dump_module()
+        ## Stored in dill.settings['dump_module']['filters'].
+        exclude.names = some_var, SomeClass
+        exclude.regexes = '_.+'
+        exclude.types = function, ModuleType, io.BytesIO
+        exclude.funcs =
+            lambda obj: type(obj.value) == int
+            dill.session.size_filter('10 KB')
+        include = _keep_this_var, '__.+__'
+
+        [filters.some.module]
+        # Filter rules specific to the module 'some.module'
+        ## Reuse regex filters defined in the previous section.
```

Comment:
?

Reply:
The parser performs substitution, `${filters:exclude.regexes}` is substituted by the `exclude.regexes` value of the `[filters]` section.